### ISSUE TITLE
SC2320 warning in function

### ISSUE THREAD

**Issue by BrentHuang** [koalaman/shellcheck#2990]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2320
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash

demo() {
    echo "in function demo..."
    echo "the script name: $0"
    echo "argument count (not include the script itself): $#"
    echo "the first argument: $1"
    echo "the second argument: $2"
    echo "the 10th argument: ${10}"

    # shellcheck disable=SC2145
    echo "argument list: $@"  # echo 输出字符串，这里 shellcheck 会告警，字符串形式用 $* 就不会告警
    echo "argument list string: $*"
    echo "process id: $$"

    ret=$?
    echo "command exited with ${ret}"
    if [[ ${ret} -ne 0 ]]; then
        echo "failed"
    fi
}

demo 1 2 3

```

#### Here's what shellcheck currently says:

In ./docs/test.sh line 16:
    ret=$?
        ^-- SC2320 (warning): This $? refers to echo/printf, not a previous command. Assign to variable to avoid it being overwritten.

#### Here's what I wanted or expected to see:

no SC2320 warning.
